### ISSUE TITLE
Keep Getting Accessory Note Found in home.app

### ISSUE THREAD

**Issue by ScottEKendall** [Ghawken/HomeKitLink-Siri#14]
Got the plugin installed, installed the security update via terminal, Indigo 2022.1 sees the plugin, was able to create the new HKLS item and went to add the QR Code to my home app and it spins forever and finally gives up and says accessory not added.

**Comment by Ghawken** [comment]
Hi -  without any further info pretty hard to be sure…

By security update - I presume you mean installing cyptography?  I presume the plugin starts without errors?
“Create new HKLS item” I presume you mean a HKLS bridge?
Have you added any accessories to the HKLS bridge? 
Do you have any error messages, or indeed any messages in the log ? Eg. Is the accessory bridge running?

This is all pretty basic low level stuff that no one else is seeing - so would presume there is some user error somewhere along the way, but without further information it is impossible to comment.

**Comment by ScottEKendall** [comment]
current setup, Mac Mini M1 (MacOS 12.4) with Indigo 2022.1.  Using a Insteon PowerLinc 2413U, Firmware 9.9.  Here is the debug log file from Indigo:

HomeKitLink Siri                Set Logging to DEBUG
   HomeKitLink Siri                Restart all Bridges Called
   HomeKitLink Siri Debug          HomeKit Bridge + Siri: Stopping homeKitBridgeDevice Device 675876799
   HomeKitLink Siri Debug          Stopping HomeKit Bridge + Siri HomeKitLink Bridge with Unique Bridge ID 702681
   HomeKitLink Siri Debug          Checkid 702681 and total deviceBridgeNumber [702681]
   HomeKitLink Siri Debug          After: Checkid 702681 and total deviceBridgeNumber []
   HomeKitLink Siri Debug          Removed Bridge <accessory display_name='HomeKitLink Bridge 702681' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          Found Driver to stop Driver Id: 702681
   HomeKitLink Siri Debug          Driver to be deleted using port 51827 and portsinUse now set()
   HomeKitLink Siri Debug          Removed Driver <pyhap.accessory_driver.AccessoryDriver object at 0x109d78910>
   HomeKitLink Siri Debug          Found Thread ID to also Stop..702681
   HomeKitLink Siri                Stopping accessory HomeKitLink Bridge 702681 on address 10.61.0.6, port 51827.
   HomeKitLink Siri                Closed the flux capacitor loop.
   HomeKitLink Siri Debug          Removed Thread object <Thread(702681, stopped daemon 6116962304)>
   HomeKitLink Siri                HomeKit Bridge + Siri has completed full Bridge, Driver and Bridge Thread Shutdown.
   HomeKitLink Siri Debug          HomeKit Bridge + Siri: Starting homeKitBridgeDevice Device 675876799 
   HomeKitLink Siri Debug          checkid = 702681 and self.deviceBridgeNumber = []
   HomeKitLink Siri Debug          Starting HomeKit Bridge + Siri HomeKitLink Bridge with Unique Bridge ID 702681
   HomeKitLink Siri Debug          create Device List Internal called
   HomeKitLink Siri Debug          Attempting to start a single Bridge:  Bridge HomeKit Bridge + Siri and ID 702681
   HomeKitLink Siri Debug          Next Port available:51827
   HomeKitLink Siri Debug          Sets of Ports Currently in Use: {51827}
   HomeKitLink Siri Debug          get_bridge_multiple called and self.device_list_internal length 3
   HomeKitLink Siri Debug          Item :{'deviceid': 1304176463, 'devicename': 'Water Heater', 'accessory': <accessory display_name='Water Heater' services=['AccessoryInformation', 'Outlet']>, 'aid': 1304176463, 'subtype': 'Outlet', 'bridgeID': 702681, 'devicemodel': 'OutletLinc', 'devicesensor': '', 'manufacturername': ''} exisits for this bridge <accessory display_name='HomeKitLink Bridge 702681' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          Adding Accessory:<accessory display_name='Water Heater' services=['AccessoryInformation', 'Outlet']>
   HomeKitLink Siri Debug          Item :{'deviceid': 868690927, 'devicename': 'Office LIght', 'accessory': <accessory display_name='Office LIght' services=['AccessoryInformation', 'Switch']>, 'aid': 868690927, 'subtype': 'Switch', 'bridgeID': 702681, 'devicemodel': 'ICON Relay Switch', 'devicesensor': '', 'manufacturername': ''} exisits for this bridge <accessory display_name='HomeKitLink Bridge 702681' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          Adding Accessory:<accessory display_name='Office LIght' services=['AccessoryInformation', 'Switch']>
   HomeKitLink Siri Debug          Item :{'deviceid': 1170788319, 'devicename': 'Thermostat', 'accessory': <accessory display_name='Thermostat' services=['AccessoryInformation', 'Thermostat']>, 'aid': 1170788319, 'subtype': 'Thermostat', 'bridgeID': 702681, 'devicemodel': 'Thermostat Adapter -2441V-', 'devicesensor': '', 'manufacturername': ''} exisits for this bridge <accessory display_name='HomeKitLink Bridge 702681' services=['AccessoryInformation', 'HAPProtocolInformation']>
   HomeKitLink Siri Debug          °C unit Selected
   HomeKitLink Siri Debug          Found Min 7 and Max 35 Temperatures 
   HomeKitLink Siri Debug          Thermostat Device using Chars: ['CurrentHeatingCoolingState', 'TargetHeatingCoolingState', 'CurrentTemperature', 'TargetTemperature', 'CoolingThresholdTemperature', 'HeatingThresholdTemperature']
   HomeKitLink Siri Debug          Adding Accessory:<accessory display_name='Thermostat' services=['AccessoryInformation', 'Thermostat']>
   HomeKitLink Siri                Scan the QR code with your HomeKit app on your iOS device:
   HomeKitLink Siri                Select Device QR button to Display QR Code, or use pincode.
   HomeKitLink Siri                Starting accessory HomeKitLink Bridge 702681 on address 10.61.0.6, port 51827.
   HomeKitLink Siri Debug          Bridge 702681 Setup and Running.
   HomeKitLink Siri                Scan the QR code with your HomeKit app on your iOS device:
   HomeKitLink Siri                Select Device QR button to Display QR Code, or use pincode.
   HomeKitLink Siri Debug          [<service display_name=AccessoryInformation chars={'Identify': False, 'Manufacturer': '', 'Model': 'OutletLinc', 'Name': 'Water Heater', 'SerialNumber': '7026811304176463', 'FirmwareRevision': ''}>, <service display_name=Outlet chars={'On': False, 'OutletInUse': True}>]
   HomeKitLink Siri Debug          [<service display_name=AccessoryInformation chars={'Identify': False, 'Manufacturer': '', 'Model': 'ICON Relay Switch', 'Name': 'Office LIght', 'SerialNumber': '702681868690927', 'FirmwareRevision': ''}>, <service display_name=Switch chars={'On': True}>]
   HomeKitLink Siri Debug          [<service display_name=AccessoryInformation chars={'Identify': False, 'Manufacturer': '', 'Model': 'Thermostat Adapter -2441V-', 'Name': 'Thermostat', 'SerialNumber': '7026811170788319', 'FirmwareRevision': ''}>, <service display_name=Thermostat chars={'CurrentHeatingCoolingState': 2, 'TargetHeatingCoolingState': 2, 'CurrentTemperature': 72.0, 'TargetTemperature': 21.0, 'TemperatureDisplayUnits': 0, 'CoolingThresholdTemperature': 23.0, 'HeatingThresholdTemperature': 19.0}>]


It appears that Indigo is publishing everything OK, but when I tried to add the accessory to a freshly (re)imaged iPhone 13, it gives me Accessory Not found after I scan in the QR code.

**Comment by Ghawken** [comment]
OK.  Well broadly it is either a plugin issue, or a HomeKit app issue.
Given the lack of any errors would be leaning towards a HomeKit app.

Although as general rule wouldn’t use the restart all Bridges menu item - isn’t needed.   Use the main restart Plugin selection - more fool proof.

But:
Is this device on the same Wifi network?  Check troubleshooters with this issue in the App in mind.
Check your Indigo computer is accessible and isn’t behind a firewall
Check whether can connect the accessory from other iOS device.

& Back to plugin.
Would remove all published devices from the bridge unselecting the publish button and pressing save for each.
& then delete bridge device.
Create new one, and add single Light only and try with this, or indeed with no published device.  
If there is no published device and the bridge is running, very unlikely to be plugin, more like connection issue within your network, this iPhone device.

& further
Could also create a new home on any current iOS device and add accessories to that.   Then can delete this ‘new home’ and move on from there.

**Comment by rubinjm** [comment]
Having the same issue--iOS Home app shows Connecting to Bridge then after a few minutes--"unable to add accessory; accessory is not reachable."

I tried removing published devices from bridge, deleted bridge device and created a new bridge device. Indigo shows it as operational. No errors in the Indigo log.

Indigo 2022.1.1
HLKS 0.3.2
MacOS 12.4

**Comment by Ghawken** [comment]
As best I can tell - this will be a HomeKit connection issue, as without errors seems unlikely to be plugin.  Obviously most others are connecting fine.

Will write a FAQ somewhere for future reference:

The plugin tests port before it uses them for the Bridge (so they should be fine).  If any question that might be port issue can in advanced settings change default port number add, minus hundred for example.  This should NOT be issue as would give error on startup.

Basically HomeKit can’t connect to your Indigo Mac.
Firewall?
Network issues?
Blocked particularly Ports?
mDNS issue?
Bonjour connection issue?
iOS device issue (old IOS devices have issues)

This is pretty generic:  But worthwhile reviewing.
https://www.macworld.com/article/673739/how-to-fix-homekit-devices-wont-connect-problems.html#:~:text=Open%20Settings%2C%20tap%20Wi%2DFi,your%20wireless%20network%20is%20down.

**Comment by Ghawken** [comment]
**Try**:
“
One common error occurs after scanning the QR code. While tryin to connect for 20 seconds or so, the following error is displayed: Home couldn't connect to this accessory.

This appears to result from some cache issue in the Home app as resetting homebridge (above) does not resolve the issue. This [WiFi trick](https://github.com/nfarina/homebridge/issues/1794) appears to work however:

Reset Pairing/Accessories as described above.
On your iOS device, use the Forget Network feature in your WiFi settings.
Rejoin your WiFi network.
Attempt to add the homebridge platforms/accessories as you normally would.

&”


This error can also occur when [mDNS] needs to be configured, e.g. when Avanti needs to be setup on Linux.

**Comment by rubinjm** [comment]
Thanks--this worked!!


> Reset Pairing/Accessories as described above.
> On your iOS device, use the Forget Network feature in your WiFi settings.
> Rejoin your WiFi network.
> Attempt to add the homebridge platforms/accessories as you normally would.

**Comment by ScottEKendall** [comment]
I did a complete reset of Indigo and used the new 0.6.5 plugin and that seemed to have fixed it...pretty sure it had something to do with the mDNS issue...

**Comment by hishamk** [comment]
I'm getting the same issue. It seems the bridges are not broadcast via mDNS. I use Avahi on my pfSense and all other services are broadcasting fine from the Indigo Mac mini. However, for the plugin, nothing seems to be broadcast (I have setup three bridges to try, still no service broadcast for the respective ports 51826, 51827, 51828). I have a _companion-link._tcp. broadcast on port 49155 but I'm not sure if that's to do with the plugin.

Any ideas?

**Comment by Ghawken** [comment]
Hi

Accessory not found is a pretty generic issue - in your case would guess another cause relating to network 

Either mDNS isn’t leaving your Mac to be seen or it is being blocked somewhere.   Companion is nothing to do with plugin 

would upgrade to the latest version - specifically
Introduces some changes for mDNS and network issues

https://github.com/Ghawken/HomeKitLink-Siri/releases/tag/0.6.16

failing the new version working off the bat would trial the mDNS settings starting with IPv4 only setting.

(these settings can break mDNS but nothing that can’t be fixed by setting back to original and restarting plugin)

**Comment by hishamk** [comment]
Thanks, I'll give the new version a try and report back.
